DigitalLearningSolutions.Data.Tests/Services/LoginServiceTests.cs

```diff
@@ -261,91 +310,147 @@ public void
-            A.CallTo(() => userService.ResetFailedLoginCount(userEntity.UserAccount)).DoesNothing();
 
             // When
-            var result = loginService.AttemptLogin(Username, Password);
+            var result = loginService.GetChooseACentreAccountViewModels(userEntity);
 
             // Then
-            using (new AssertionScope())
-            {
-                result.LoginAttemptResult.Should().Be(LoginAttemptResult.ChooseACentre);
-                result.UserEntity.Should().Be(userEntity);
-                result.CentreToLogInto.Should().BeNull();
-            }
+            result.Should().BeEquivalentTo(
+                new List<ChooseACentreAccountViewModel>
+                {
+                    new ChooseACentreAccountViewModel(2, "delegate", true, false, true, true, true),
+                }
+            );
         }
 
         [Test]
-        public void
-            AttemptLogin_returns_choose_a_centre_when_user_account_has_no_centres_and_successfully_logging_in()
+        public void GetChooseACentreAccountViewModels_returns_empty_list_when_only_inactive_admin_accounts_exist()
+        {
+            // Given
+            var userEntity = new UserEntity(
+                UserTestHelper.GetDefaultUserAccount(),
+                new List<AdminAccount> { UserTestHelper.GetDefaultAdminAccount(active: false) },
+                new List<DelegateAccount>()
+            );
+
+            // When
+            var result = loginService.GetChooseACentreAccountViewModels(userEntity);
+
+            // Then
+            result.Should().HaveCount(0);
+        }
+
+        [Test]
+        public void GetChooseACentreAccountViewModels_returns_empty_list_when_no_admin_or_delegate_accounts()
         {
             // Given
             var userEntity = new UserEntity(
                 UserTestHelper.GetDefaultUserAccount(),
                 new List<AdminAccount>(),
                 new List<DelegateAccount>()
             );
-            A.CallTo(() => userService.GetUserByUsername(Username)).Returns(userEntity);
-            A.CallTo(() => userVerificationService.VerifyUserEntity(Password, userEntity))
-                .Returns(new UserEntityVerificationResult(true, new List<int>(), new[] { 2 }, new List<int>()));
-            A.CallTo(() => userService.ResetFailedLoginCount(userEntity.UserAccount)).DoesNothing();
 
             // When
-            var result = loginService.AttemptLogin(Username, Password);
+            var result = loginService.GetChooseACentreAccountViewModels(userEntity);
 
             // Then
-            using (new AssertionScope())
-            {
-                result.LoginAttemptResult.Should().Be(LoginAttemptResult.ChooseACentre);
-                result.UserEntity.Should().Be(userEntity);
-                result.CentreToLogInto.Should().BeNull();
-            }
+            result.Should().HaveCount(0);
         }
     }

```

Comment:
Could also be good to test for AttemptLogin() that
* the ChooseACentre login result is returned if the user has accounts at more than one centre and they've logged in using email
* the direct login result is returned if the user logs in using the candidate number of an available account.

Reply:
Done